### PR TITLE
Powering HackIllinois

### LINKED ISSUE #30: API, Infrastructure, and Technical Posts
Post Ideas
-----------
- How we redesigned out API to use microservices and rewrote in in Go
- The infrastructure we run on AWS and our deployment pipeline

These are just a couple general ideas that we can work with for some of the more technical blog posts. If you have any other ideas add them in here.

### PR DESCRIPTION
**Pull request by ASankaran**
Closes #30.

This post is about the HackIllinois API and some the decisions that were made while designing and implementing it.

## Pre-Approval
- [ ] Post properly titled
- [ ] Post contains proper front matter
- [ ] `featured` tag has been added to this post
- [ ] `featured` tag has been removed from other posts
- [ ] Images and assets have been added
- [ ] Post has been tested locally with Jekyll
- [ ] Spelling has been checked

## Pre-Deploy
- [ ] Reviews have been resolved or otherwise addressed
- [ ] Approval from a repo owner
- [ ] Post file name has been updated to current date
- [ ] Post front matter has been updated to current date
- [ ] Merged master

### CODE REVIEW

**Comment by apache8080** [comment]
I think most of the post is really solid. The individual section on each service may not be necessary for this post. If there were some services that had interesting challenges then we may be able to spin it out into another post.

**Comment by justinyangusa** [comment]
Great start, @ASankaran! Based on our conversation yesterday, I’ll be looking for the next revision to be shorter with additional details on the microservice architecture decision. Descriptions for each microservice can be moved into documentation for the API.

File: _posts/2018-08-11-powering-hackillinois.markdown [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	- Allow staff to track which hackers have taken part in events
+There are a certainly more features which would be very useful for both staff  and hackers at HackIllinois, but these are the core requirements we will start off with. Once completed, we can explore more advanced functionality.
+
+### Implementation Strategy
+As mentioned at the begining of the post, there have been a few iterations of the API over the years. The purpose and requirements of the API over the years has remainds relativley constant, however the implementation has greatly different from iteration to iteration. Each iteration of the API aims to innovate and take a new approach to solving problems which arose in the past. [Last year's API](https://github.com/HackIllinois/api-2018) was implemented as a monolithic application using nodejs and MySQL for the database. It worked and cewrtainl met the requirements defined for HackIllinois 2018, however some issue we ran into during the development was the lack of compile time type safety and annoyance of building a new database migration for any minor change to the defined scheme. This year we aimed to reimplement the API from scratch and resolve these issues.
```
Comment (redsn0w422) [inline _posts/2018-08-11-powering-hackillinois.markdown]: spellcheck

**Review by redsn0w422 — CHANGES_REQUESTED** [review]
Good start! Remember to spell check the post 😄 

Also, I'd be more explicit about why we opted to write our own API vs. using something else, or something more simple like Typeform/Google docs/etc.

**Pull request closed without merging**